### DISCUSSION TITLE
Error when reading tables using dblink LONG type

### DISCUSSION THREAD

**Discussion by FelipeCardos0** [oracle/python-oracledb#213] [Q&A]
I'm trying to read a table that has a dblink configured. 
This table has two columns: id NUMBER(10) , data LONG. The LONG type column has data greater than 32k.
But every time I try to read the table with this column I get the following error:  DPY-3006: Oracle data type 2 is not supported.

Please find more details below
Thank you for your help

VERSION
```
Oracle Database 19c Enterprise Edition Release 19.0.0.0.0
python==3.10.12
oracledb==1.3.2
```

Error
```
DatabaseError: Execution failed on sql 'select * from XXX ': DPY-3006: Oracle data type 2 is not supported
```

Code Python
```
import oracledb
con = oracledb.connect(f"...")

cur = con.cursor()
cur.execute( """select "id","data" from dblink.XXXX  """ )
cur.close()
con.close()
```

Oracle Table Detail

```
Name         Null?    Type       
------------ -------- ---------- 
id                      NUMBER(10)     
data                    LONG       

```

**Comment by anthony-tuininga** [comment]
Are you saying that you have a table that is on the other side of a DB link that has the details you mentioned? And you are trying to fetch the contents of that table? That normally looks like this:

```
cur.execute("select id, data from mytable@mydblink")
```

Can you clarify?

**Comment by FelipeCardos0** [comment]
I understand your doubt.
But I don't know how the dblink was created. The information was only made available in this way.
I know that the source table is a mysql database, and we access it via dblink as I indicated above.

**Reply by anthony-tuininga** [reply]
So the dblink is to a *MySQL* database? Can you find out how the dblink was created? And can you provide any further details that might help replicate the issue? From the SQL it looks to me like you are not dealing with a dblink since you are missing the "@" character in your SQL statement. Please clarify!

**Comment by cjbj** [comment]
@FelipeCardos0 as a workaround, have you tried Thick mode? See https://python-oracledb.readthedocs.io/en/latest/user_guide/initialization.html#enabling-python-oracledb-thick-mode
